pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -88,6 +109,35 @@ func diagnose(diagCfg diagnosis.Config, _ sender.DiagnoseSenderManager) []diagno
 	return diagnoses
 }
 
+func createDiagnosis(name string, logURL string, report string, err error) diagnosis.Diagnosis {
+	d := diagnosis.Diagnosis{
+		Name: name,
+	}
+
+	if err == nil {
+		d.Result = diagnosis.DiagnosisSuccess
+		diagnosisWithoutReport := fmt.Sprintf("Connectivity to `%s` is Ok", logURL)
+		d.Diagnosis = createDiagnosisString(diagnosisWithoutReport, report)
+	} else {
+		d.Result = diagnosis.DiagnosisFail
+		diagnosisWithoutReport := fmt.Sprintf("Connection to `%s` failed", logURL)
+		d.Diagnosis = createDiagnosisString(diagnosisWithoutReport, report)
+		d.Remediation = "Please validate Agent configuration and firewall to access " + logURL
+		d.RawError = err.Error()
+	}
+
+	return d
+}
+
+func createDiagnosisString(diagnosis string, report string) string {
+	if len(report) == 0 {
+		return diagnosis
+	}
+
+	diagnosis = strings.Join([]string{diagnosis, "\n", report}, "")
```

Comment:
```suggestion
	return strings.Join([]string{diagnosis, "\n", report}, "")
```

NIT: Can just return here